gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -189,14 +251,141 @@ var gpii = fluid.registerNamespace("gpii");
     });
 
     /**
-     * Manually constructs the functions in the queueFunctions options.
+     * Function to check whether a proximity trigger should bounce.
+     * Debounce rule: any RFID actions is ignored for <myGpiiKey> if a login/logout for <myGpiiKey> is in progress
+     *                OR if the last login/logout process for <myGpiiKey> finished less than 1.5 seconds ago
+     * If no user is logged in and debounce doesn't apply, log in <myGpiiKey>
+     * If <myGpiiKey> is logged in and debounce doesn't apply, log out <myGpiiKey>
+     * If another user is already logged in or in the process of logging in or out, log that user out and log in <myGpiiKey>
+     *
+     * @param {Component} that - An instance of gpii.lifecycleManager.
+     * @param {Integer} debounceTimeMs - The time in milliseconds for which to debounce after last proximityTrigger was fired.
+     * @param {String} gpiiKey - The GPII key of the user for whom the proximityTrigger was fired.
+     * @return {Boolean} Returns true if the proximityTrigger should be ignored (i.e. is considered a bounce). This is done according
+     *              to the following rule: any RFID actions is ignored for <myGpiiKey> if a login/logout for <myGpiiKey> is in progress
+     *              OR if the last login/logout process for <myGpiiKey> finished less than {debounceTimeMs} milliseconds ago
+     */
+    gpii.lifecycleManager.isBounce = function (that, debounceTimeMs, gpiiKey) {
```

Comment:
This only reads currentLogonChange and currentUserLogon so will be clearer if these are just surfaced as explicit arguments. It will make the function shorter as well!